### ISSUE TITLE
Make this repo a template repo

### ISSUE THREAD

**Issue by tobiasdiez** [nuxt/starter#247]
https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository

so that one could easily start a new module etc by simply using the "Create repo from template" option.

**Comment by TheAlexLichter** [comment]
That might be a bit difficult because the starter repo contains several starter templates.

**Comment by tobiasdiez** [comment]
Yes, that's a bit of a disadvantage of the current organization. However, when creating a repo from a template, one can copy all branches (see point 7 in https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template). Afterwards, one could rename the branch one wants to `main` and make it the default branch, then delete all branches one doesn't need.

**Comment by atinux** [comment]
Don't you think it is a bit longer than creating the repo on GitHub -> Clone it -> nuxi init X -> git push?

Otherwise we may have to split each branch to it's own repo.

Or.... give the possibility on `nuxt.new` [nuxt/nuxt.new](https://github.com/nuxt/nuxt.new) to connect with GitHub (OAuth) to create a repository from a template 👀

**Comment by TheAlexLichter** [comment]
I like the GitHub connection idea! 👏🏻

**Comment by atinux** [comment]
Good news, the nuxt.new repository is open source and open to anymore to contribute 😊 

I made https://github.com/Atinux/discuss to show how to make a Nuxt 3 app with GitHub oauth if it can help.

**Comment by tobiasdiez** [comment]
The main advantage of using github templates over `nuxi init` is that one gets the git data with it, so that's relatively straightforward to pull improvements in the starter repo. Another small plus is that github would write "generated from https://github.com/nuxt/starter", which gives free advertisement to the starter repo.

But I agree that there are not many big advantages. 

> Otherwise we may have to split each branch to it's own repo.

This sounds like the most viable/easy approach.
